src/categorical_algebra/Slices.jl

```diff
@@ -0,0 +1,133 @@
+module Slices
+export Slice, SliceMorphism,SliceLimit,SliceColimit, pushout_complement
+
+using AutoHashEquals
+using ...GAT
+using ..FreeDiagrams, ..Limits, ..CSets
+using ...Theories: Category
+import ...Theories: dom, codom, compose, id
+import ..Limits: limit, colimit, universal
+import ..FinSets: force, pushout_complement
+
+@auto_hash_equals struct Slice{Ob,Hom}
```

Comment:
From the PoV of dependent types, I understand the temptation to carry around the extra type parameter `Ob`, but ultimately I think we shouldn't because it's not in line with Julia's philosophy on types: it's not needed for dispatch or struct definition, and it's just extra stuff that has to be manually supplied and carried around.

Then you should also add a second constructor:

```julia
Slice(f::Hom) where Hom = Slice{Hom}(f)
```